src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.cs

```diff
@@ -97,11 +85,33 @@ private ImmutableArray<CodeAction> CreateActions(State state, CancellationToken
             return actions.ToImmutableArray();
         }
 
-        private CodeAction GetCodeAction(
-            State state,
-            OperationKind operationKind)
+        private CodeAction GetCodeAction(State state, OperationKind operationKind) =>
+            new MoveTypeCodeAction((TService)this, state, operationKind);
+
+        private bool IsNestedType(TTypeDeclarationSyntax typeNode) =>
+            typeNode.Parent is TTypeDeclarationSyntax;
+
+        /// <summary>
+        /// checks if there is a single top level type declaration in a document
+        /// </summary>
+        private bool MultipleTopLevelTypeDeclarationInSourceDocument(SyntaxNode root) =>
+            TopLevelTypeDeclarations(root).Count() > 1;
+
+        private static IEnumerable<TTypeDeclarationSyntax> TopLevelTypeDeclarations(SyntaxNode root) =>
+            root.DescendantNodes(n => (n is TCompilationUnitSyntax || n is TNamespaceDeclarationSyntax))
+                .OfType<TTypeDeclarationSyntax>();
+
+        private static bool AnyTopLevelTypeMatchesDocumentName(State state, CancellationToken cancellationToken)
         {
-            return new MoveTypeCodeAction((TService)this, state, operationKind);
+            var root = state.SemanticDocument.Root;
+            var semanticModel = state.SemanticDocument.SemanticModel;
+
+            return TopLevelTypeDeclarations(root).Any(
+                typeDeclaration => 
+                {
+                    var typeName = semanticModel.GetDeclaredSymbol(typeDeclaration, cancellationToken).Name;
+                    return string.Equals(state.DocumentName, typeName, StringComparison.CurrentCulture);
```

Comment:
Doesn't this mean that i fi have "Outer.Inner.cs" and you'll prompt to rename this file to "Outer.cs"?